woocommerce/woocommerce-gutenberg-products-block
# Add button alignment option to product grid blocks

This PR adds a new toggle option to align product add to cart buttons vertically. I've implemented this across all grid based blocks. Tests are passing locally.


Closes #384



#### Accessibility

I tested toggling on/off the option via keyboard controls. I've followed the same pattern as other toggles found in the visibility section.


### Screenshots

This is how it looks toggled on:
![Edit Post ‹ local wordpress test — WordPress 2019-05-28 17-19-55](https://user-images.githubusercontent.com/90977/58494370-dadedb00-816c-11e9-9d45-cfda0b5243f2.png)

And toggled off (default):
![Edit Post ‹ local wordpress test — WordPress 2019-05-28 17-20-26](https://user-images.githubusercontent.com/90977/58494391-e500d980-816c-11e9-8a87-58f1720d7a46.png)

When enabled, product cart buttons align like this using flexbox:
![Edit Post ‹ local wordpress test — WordPress 2019-05-28 17-20-52](https://user-images.githubusercontent.com/90977/58494428-fd70f400-816c-11e9-8ea1-cd8d9376b127.png)

### How to test the changes in this Pull Request:

1. For each of the grid based block types, ensure you can see this option under "layout", toggle the option, and that the option stays toggled on after update and refresh.
2. Check the flexbox works by setting up products with varying title lengths. Buttons should be aligned like the screenshot above.
3. Check (2) on the frontend too - same styling should apply there.
